samples/nrf9160/https_simple/src/main.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <net/socket.h>
+#include <stdio.h>
+#include <uart.h>
+#include <string.h>
+
+#define HTTP_HOST "google.com"
+#define HTTP_PORT 443
+#define RECV_BUF_SIZE 4096
+
+char recv_buf[RECV_BUF_SIZE + 1];
+
+void app_https_start(void)
+{
+	struct addrinfo *res;
+	struct addrinfo hints;
+
+	hints.ai_flags = 0;
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_protocol = 0;
+
+	printk("HTTPS simple example\n\r");
+
+	int err = getaddrinfo(HTTP_HOST, NULL, &hints, &res);
+
+	if (err) {
+		printk("getaddrinfo errno %d\n", errno);
+		/* No clean up needed, just return */
+		return;
+	}
+
+	((struct sockaddr_in *)res->ai_addr)->sin_port = htons(HTTP_PORT);
+
+	char send_buf[] =
+		"GET / HTTP/1.1\r\n"
+		"Host: www.google.com:443\r\n"
+		"Connection: close\r\n\r\n";
+	int send_data_len = strlen(send_buf);
+
+	int client_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TLS_1_2);
+
+	enum {
+		NONE = 0,
+		OPTIONAL = 1,
+		REQUIRED = 2,
+	};
+
+	int verify = OPTIONAL;
+
+	err = setsockopt(client_fd, SOL_TLS, TLS_PEER_VERIFY, &verify,
```

Comment:
I wonder if we should do this. It's a sample, we should show best practices here - and ignoring certificate verification result is not a good one. Why don't we provision one of the CA root certificates used by google and do proper TLS?

Reply:
I want this to be as small as possible, with little overhead. Adding nrf_inbuilt etc, with root cert, is something we do in other samples, and that is not what this sample is made to show. I can ofcourse add a comment that you should issue a root CA, and verify your hostname.

Reply:
I'm still standing in a position, that if we want to show TLS, we should do it right. If we want to keep the sample small, we should do HTTP instead.

My point is, that these samples are often copied 'as is', even internally. If we show in a sample a non-secure way to establish a TLS connection, which bypasses authentication (and should only be used during development/debugging), it will sooner or later end up in other places as well.

Reply:
I would be happy to have a sample to show our customers how to provision keys with that lib and setup a secure connection.